main.go

```diff
@@ -1,31 +1,93 @@
 package main
 
 import (
+	"flag"
 	"fmt"
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/credentials"
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/elb"
+	etcdClient "github.com/coreos/etcd/client"
+	etcdContext "golang.org/x/net/context"
 	"log"
+	"net/http"
 	"os"
+	"strings"
+	"time"
 )
 
-func main() {
+const (
+	konsDNSEndPoint = "https://konstructor.ft.com/v1/dns/"
+)
+
+var (
+	etcdPeers = flag.String("etcdPeers", "", "Comma-separated list of addresses of etcd endpoints to connect to")
+)
 
-	region := os.Getenv("AWS_REGION")
-	if region == "" {
-		log.Fatal("missing AWS_REGION environment variable")
+type conf struct {
+	env             string
+	konsAPIKey      string
+	konsDNSEndPoint string
+	elbName         string
+	awsAccessKey    string
+	awsSecretKey    string
+	awsRegion       string
+}
+
+func set(kapi etcdClient.KeysAPI, s *string, keyName string, e *error) {
+	var resp *etcdClient.Response
+	if *e != nil {
+		return
+	}
+	resp, *e = kapi.Get(etcdContext.Background(), keyName, nil)
+	if *e != nil {
+		return
 	}
+	*s = resp.Node.Value
+}
+
+func config() *conf {
+	var (
+		err error
+		c   conf
+	)
 
-	elbName := os.Getenv("ELB_NAME")
-	if elbName == "" {
-		log.Fatal("missing ELB_NAME environment variable")
+	cfg := etcdClient.Config{
+		Endpoints:               strings.Split(*etcdPeers, ","),
+		HeaderTimeoutPerRequest: 10 * time.Second,
 	}
 
+	etcd, err := etcdClient.New(cfg)
+	if err != nil {
+		log.Fatal(err)
+	}
+	kapi := etcdClient.NewKeysAPI(etcd)
+
+	set(kapi, &c.env, "/ft/config/environment_tag", &err)
+	set(kapi, &c.konsAPIKey, "/ft/_credentials/konstructor/api-key", &err)
+	set(kapi, &c.elbName, "/ft/_credentials/elb_name", &err)
+	set(kapi, &c.awsAccessKey, "/ft/_credentials/aws/aws_access_key_id", &err)
+	set(kapi, &c.awsSecretKey, "/ft/_credentials/aws/aws_secret_access_key", &err)
+	set(kapi, &c.awsRegion, "/ft/config/aws_region", &err)
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	c.konsDNSEndPoint = konsDNSEndPoint
+
+	return &c
+}
+
+func elbDNSName(c *conf) {
+	//weirdness around how aws handles credentials
+	os.Setenv("AWS_ACCESS_KEY_ID", c.awsAccessKey)
+	os.Setenv("AWS_SECRET_ACCESS_KEY", c.awsSecretKey)
+
 	svc := elb.New(
 		session.New(
 			&aws.Config{
-				Region:      aws.String(region),
+				Region:      aws.String(c.awsRegion),
 				Credentials: credentials.NewEnvCredentials(),
```

Comment:
you can use something other than NewEnvCredentials to avoid the Setenv stuff above.  I think we did this for something else? (elb name getter or whatever??)